test/livebook_web/live/session_live_test.exs

```diff
@@ -949,11 +949,51 @@ defmodule LivebookWeb.SessionLiveTest do
 
       view
       |> element(~s{form[phx-submit="save"]})
-      |> render_submit(%{data: %{name: "foo", value: "123"}})
+      |> render_submit(%{data: %{name: "foo", value: "123", store: "session"}})
 
       assert %{secrets: [%{name: "FOO", value: "123"}]} = Session.get_data(session.pid)
     end
 
+    test "adds a livebook secret from form", %{conn: conn, session: session} do
+      {:ok, view, _} = live(conn, "/sessions/#{session.id}/secrets")
+
+      view
+      |> element(~s{form[phx-submit="save"]})
+      |> render_submit(%{data: %{name: "bar", value: "456", store: "livebook"}})
+
+      assert %Livebook.Secrets.Secret{name: "BAR", value: "456"} in Livebook.Secrets.fetch_secrets()
+    end
+
+    test "sync secrets when they're equal", %{conn: conn, session: session} do
+      {:ok, view, _} = live(conn, "/sessions/#{session.id}/secrets")
+      Session.put_secret(session.pid, %{name: "FOO", value: "123"})
+      Livebook.Secrets.set_secret(%{name: "FOO", value: "123"})
+
+      view
+      |> element(~s{form[phx-submit="save"]})
+      |> render_submit(%{data: %{name: "FOO", value: "456", store: "session"}})
+
+      assert %{secrets: [%{name: "FOO", value: "456"}]} = Session.get_data(session.pid)
+
+      assert %Livebook.Secrets.Secret{name: "FOO", value: "456"} in Livebook.Secrets.fetch_secrets()
+    end
```

Comment:
Instead of relying on this heuristic, perhaps we should add something like `:source` to session secrets, so we can differentiate? @josevalim @cristineguadelupe

Reply:
@jonatanklosko I don't think I get the idea 🤔